_docs/how-to/python-package.md

```diff
@@ -0,0 +1,207 @@
+
+We don’t recommend using the `ops` or `charm` namespace for your charm libraries distributed as Python packages.
+It will be easier for charmers to follow your code if the `ops` namespace is reserved for the `ops` package.
+Likewise, the `charms` namespace is best left for charmcraft managed libs.
+
+If your library is only intended to be used by your own charms,
+for example if you write a library to be used by the machine and Kubernetes versions of your charm,
+releasing a package on PyPI using this naming scheme may not be the right approach,
+as you don't necessarily want to advertise the library for public consumption.
+
+(python-package-distribution)=
+## How to distribute your Python package
+
+Distributing your package on PyPI has the most benefits.
+However, you may find it useful to begin by distributing your package via a git url during development and internal use.
+Using a git dependency
+or skipping distribution in favour of packing the local package
+may be appropriate if your library is purely for your own charms,
+and is not intended for external users.
+
+(python-package-distribution-pypi)=
+### PyPI
+
+Use [trusted publishing](https://docs.pypi.org/trusted-publishers/) to publish directly from your github repository.
+You can use the [actions/attest-build-provenance](https://github.com/actions/attest-build-provenance)
+and [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) actions in your workflows.
+Trigger the workflow based on a version tag being pushed.
+Feel free to check out this project's workflows for an example.
+Make sure that your repository only allows trusted contributors write access.
+
+The team manager and another truster team member should be the package owners on PyPI,
+using their Canonical email addresses.
+Make sure to also claim your package on [Test PyPI](https://test.pypi.org/),
+and setup a workflow for publishing there.
+All team members can be owners on Test PyPI.
+
+A major benefit of publishing on PyPI is that users of your library can specify version ranges in their dependencies.
+Therefore, if you’re going to publish on PyPI, we highly recommend that you use semantic versioning for your library.
+
+A non-dev/alpha/beta/etc qualified 1.x release to PyPI signifies that your library is ready for public consumption.
+You should also communicate this through the ["Development Status" Trove classifier](https://pypi.org/classifiers/) in your `pyproject.toml`.
+
+(python-package-distribution-git)=
+### Git
+
+You can get started by distributing your library as a Python package with very little friction using GitHub.
+This is good for prototyping, or when first transitioning from a charm-lib to a Python package,
+and may be a good fit for libraries that are intended for team-internal use.
+Once you’re done prototyping and the library is ready for external users, you’ll want to start using PyPI instead.
```

Comment:
I would weaken or remove this, because there's still a pretty reasonable case that team-internal libraries just stay on GitHub. We can present the trade-offs (you list many below regarding versioning) and let them make the call.